openid_connect.js

```diff
@@ -255,6 +250,13 @@ function validateIdToken(r) {
             r.error("OIDC ID Token validation error: nonce from token (" + r.variables.jwt_claim_nonce + ") does not match client (" + client_nonce_hash + ")");
             validToken = false;
         }
+    } else if (!r.variables.refresh_token || r.variables.refresh_token == "-") {
+        r.error("OIDC ID Token validation error: nonce absent in first token of the session");
```

Comment:
I'd suggest considering this error message:
```suggestion
        r.error("OIDC ID Token validation error: missing nonce claim in ID Token during initial authentication.");
```